### PR TITLE
Spark: Add new entry point for actions

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR adds new entry points for the redesigned actions API in Spark.

### CODE REVIEW

**Comment by aokolnychyi** [comment]
@rdblue @RussellSpitzer @openinx, could you take a look?

**Comment by aokolnychyi** [comment]
The idea is to offer new entry points like this:

```
org.apache.iceberg.spark.actions.SparkActions
org.apache.iceberg.flink.actions.FlinkActions
```

File: spark/src/main/java/org/apache/iceberg/spark/actions/BaseSparkActions.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+package org.apache.iceberg.spark.actions;
+
+import org.apache.iceberg.Table;
+import org.apache.iceberg.actions.ActionsProvider;
+import org.apache.iceberg.actions.ExpireSnapshots;
+import org.apache.iceberg.actions.RemoveOrphanFiles;
+import org.apache.iceberg.actions.RewriteManifests;
+import org.apache.spark.sql.SparkSession;
+
+abstract class BaseSparkActions implements ActionsProvider {
```
Comment (aokolnychyi) [inline spark/src/main/java/org/apache/iceberg/spark/actions/BaseSparkActions.java:29]: We don't have data compaction yet. That's why I did not deprecate the old entry points yet.

File: spark2/src/main/java/org/apache/iceberg/spark/actions/SparkActions.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.apache.iceberg.actions.ActionsProvider;
+import org.apache.spark.sql.SparkSession;
+
+/**
+ * An implementation of {@link ActionsProvider} for Spark.
+ * <p>
+ * This class is the primary API for interacting with actions in Spark that users should use
+ * to instantiate particular actions.
+ */
+public class SparkActions extends BaseSparkActions {
```
Comment (aokolnychyi) [inline spark2/src/main/java/org/apache/iceberg/spark/actions/SparkActions.java:31]: No more reflection.

**Review by rdblue — APPROVED** [review]
Looks good to me!

**Pull request merged**